x-pack/qa/rolling-upgrade/src/test/java/org/elasticsearch/upgrades/TokenBackwardsCompatibilityIT.java

```diff
@@ -14,197 +14,365 @@
 import org.elasticsearch.client.ResponseException;
 import org.elasticsearch.client.RestClient;
 import org.elasticsearch.test.rest.yaml.ObjectPath;
+import org.junit.After;
+import org.junit.Before;
 
 import java.io.IOException;
 import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
 
 public class TokenBackwardsCompatibilityIT extends AbstractUpgradeTestCase {
 
-    public void testGeneratingTokenInOldCluster() throws Exception {
+    private Collection<RestClient> twoClients = null;
+
+    @Before
+    private void collateClientsByVersion() throws IOException {
+        Map<Version, RestClient> clientsByVersion = getRestClientByVersion();
+        if (clientsByVersion.size() == 2) {
+            // usual case, clients have different versions
+            twoClients = clientsByVersion.values();
+        } else {
+            assert clientsByVersion.size() == 1 : "A rolling upgrade has a maximum of two distinct node versions, found: "
+                    + clientsByVersion.keySet();
+            // tests assumes exactly two clients to simplify some logic
```

Comment:
We only need 2 clients in mixed clusters right? Wondering if a `getCurrentClient()` and a `getOldClient()` that can be called from the tests when needed would make more sense ? Just an idea

Reply:
I think it's easier to have them in an array because we can iterate over them.
The operations are always the same irrespective of the client version.